api/comment_threads.rb

```diff
@@ -27,9 +27,12 @@
     error 404, [t(:requested_object_not_found)].to_json
   end
 
-  if params["user_id"] and bool_mark_as_read
+  if params["user_id"]
     user = User.only([:id, :username, :read_states]).find_by(external_id: params["user_id"])
-    user.mark_as_read(thread) if user
+  end
+
+  if user and bool_mark_as_read
+    user.mark_as_read(thread)
   end
```

Comment:
@wajeeha-khalid Can you add a comment in this code to make it clear that we want to assign the user even if `bool_mark_as_read` is False, so that user-specific fields, such as "read" is subsequently returned?